com.hazelcast.jet.core.ManualRestartTest.when_autoScalingDisabled_then_jobRestarts

_master_ 97f2654f0114ae8a5f7463d1e95cf4f06572e533

http://jenkins.hazelcast.com/view/Jet/job/jet-oss-master-sonar-nightly/146/

Stacktrace:
```
Error Message
expected:<6> but was:<2>
Stacktrace
java.lang.AssertionError: expected:<6> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.jet.core.ManualRestartTest.lambda$testJobRestartWhenJobIsRunning$2(ManualRestartTest.java:107)
	at com.hazelcast.jet.core.JetTestSupport$1.run(JetTestSupport.java:196)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1310)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1327)
	at com.hazelcast.jet.core.JetTestSupport.assertTrueEventually(JetTestSupport.java:161)
	at com.hazelcast.jet.core.ManualRestartTest.testJobRestartWhenJobIsRunning(ManualRestartTest.java:107)
	at com.hazelcast.jet.core.ManualRestartTest.when_autoScalingDisabled_then_jobRestarts(ManualRestartTest.java:85)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

The migration took longer than usual and the 10-second timeout wasn't enough. Excerpts from log:
```
13:06:39,091  INFO || - [MigrationManager] hz._hzInstance_26_jet.migration - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Re-partitioning cluster data... Migration queue size: 537
13:06:39,137  WARN || - [MigrationManager] hz._hzInstance_26_jet.migration - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Failed migration from Member [127.0.0.1]:5703 - b564a6a6-31ea-4224-b7b5-f7ea19748331 for MigrationInfo{uuid=b61cf2a9-6f8c-400d-97d8-80f7e3c37a5d, partitionId=0, source=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5704 - c60492b7-92c2-4958-9878-5aca0b86c82d, destinationCurrentReplicaIndex=-1, destinationNewReplicaIndex=3, master=[127.0.0.1]:5701, initialPartitionVersion=553, partitionVersionIncrement=1, processing=false, status=ACTIVE}
java.util.concurrent.ExecutionException: com.hazelcast.spi.exception.TargetNotMemberException: Destination of migration could not be found! => com.hazelcast.internal.partition.operation.MigrationRequestOperation{serviceName='hz:core:partitionService', identityHash=169587654, partitionId=0, replicaIndex=0, callId=71, invocationTime=1560344799091 (2019-06-12 13:06:39.091), waitTimeout=-1, callTimeout=300000, migration=MigrationInfo{uuid=b61cf2a9-6f8c-400d-97d8-80f7e3c37a5d, partitionId=0, source=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5704 - c60492b7-92c2-4958-9878-5aca0b86c82d, destinationCurrentReplicaIndex=-1, destinationNewReplicaIndex=3, master=[127.0.0.1]:5701, initialPartitionVersion=553, partitionVersionIncrement=1, processing=false, status=ACTIVE}}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:163)
	at com.hazelcast.internal.partition.impl.MigrationManager$MigrateTask.executeMigrateOperation(MigrationManager.java:1105)
	at com.hazelcast.internal.partition.impl.MigrationManager$MigrateTask.run(MigrationManager.java:1003)
	at com.hazelcast.internal.partition.impl.MigrationThread.processTask(MigrationThread.java:122)
	at com.hazelcast.internal.partition.impl.MigrationThread.doRun(MigrationThread.java:98)
	at com.hazelcast.internal.partition.impl.MigrationThread.run(MigrationThread.java:67)
Caused by: com.hazelcast.spi.exception.TargetNotMemberException: Destination of migration could not be found! => com.hazelcast.internal.partition.operation.MigrationRequestOperation{serviceName='hz:core:partitionService', identityHash=169587654, partitionId=0, replicaIndex=0, callId=71, invocationTime=1560344799091 (2019-06-12 13:06:39.091), waitTimeout=-1, callTimeout=300000, migration=MigrationInfo{uuid=b61cf2a9-6f8c-400d-97d8-80f7e3c37a5d, partitionId=0, source=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5704 - c60492b7-92c2-4958-9878-5aca0b86c82d, destinationCurrentReplicaIndex=-1, destinationNewReplicaIndex=3, master=[127.0.0.1]:5701, initialPartitionVersion=553, partitionVersionIncrement=1, processing=false, status=ACTIVE}}
	at com.hazelcast.internal.partition.operation.BaseMigrationOperation.verifyExistingDestination(BaseMigrationOperation.java:190)
	at com.hazelcast.internal.partition.operation.BaseMigrationOperation.verifyMigrationParticipant(BaseMigrationOperation.java:162)
	at com.hazelcast.internal.partition.operation.BaseMigrationOperation.beforeRun(BaseMigrationOperation.java:90)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:197)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:416)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:153)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:123)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:126)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79)
	... 6 more
13:06:39,138  WARN || - [MigrationManager] hz._hzInstance_26_jet.migration - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Migration failed: MigrationInfo{uuid=b61cf2a9-6f8c-400d-97d8-80f7e3c37a5d, partitionId=0, source=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5704 - c60492b7-92c2-4958-9878-5aca0b86c82d, destinationCurrentReplicaIndex=-1, destinationNewReplicaIndex=3, master=[127.0.0.1]:5701, initialPartitionVersion=553, partitionVersionIncrement=1, processing=false, status=ACTIVE}
13:06:39,558  INFO || - [LifecycleService] when_autoScalingDisabled_then_jobRestarts - [127.0.0.1]:5704 [jet] [3.1-SNAPSHOT] [127.0.0.1]:5704 is STARTED
13:06:42,560 DEBUG || - [Job] when_autoScalingDisabled_then_jobRestarts - hz.client_2 [jet] [3.1-SNAPSHOT] [3.12.1-SNAPSHOT] Sending RESTART_GRACEFUL request for job 069c-611d-30b2-6555 (name ??)
13:06:42,630 DEBUG || - [JobExecutionService] hz._hzInstance_26_jet.jet.cooperative.thread-58 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Execution of job '069c-611d-30b2-6555', execution 9cc8-1876-5004-f6b2 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: RESTART_FORCEFUL
Caused by: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: RESTART_FORCEFUL
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:195)
13:06:42,674 DEBUG || - [MasterJobContext] hz._hzInstance_26_jet.async.thread-71 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Rescheduling restart of '069c-611d-30b2-6555': cluster is not safe
13:06:44,678 DEBUG || - [MasterJobContext] hz._hzInstance_26_jet.cached.thread-8 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Rescheduling restart of '069c-611d-30b2-6555': cluster is not safe
13:06:46,701 DEBUG || - [MasterJobContext] hz._hzInstance_26_jet.cached.thread-18 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Rescheduling restart of '069c-611d-30b2-6555': cluster is not safe
13:06:48,060  INFO || - [InternalPartitionService] hz._hzInstance_26_jet.cached.thread-18 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Remaining migration tasks in queue => 190. (lastRepartitionTime=Wed Jun 12 13:06:42 UTC 2019, completedMigrations=349, totalCompletedMigrations=355, elapsedMigrationTime=5653ms, totalElapsedMigrationTime=7338ms)
13:06:48,704 DEBUG || - [MasterJobContext] hz._hzInstance_26_jet.cached.thread-17 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Rescheduling restart of '069c-611d-30b2-6555': cluster is not safe
13:06:50,719 DEBUG || - [MasterJobContext] hz._hzInstance_26_jet.cached.thread-17 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Rescheduling restart of '069c-611d-30b2-6555': cluster is not safe
13:06:52,390  INFO || - [MigrationThread] hz._hzInstance_26_jet.migration - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] All migration tasks have been completed. (lastRepartitionTime=Wed Jun 12 13:06:42 UTC 2019, completedMigrations=537, totalCompletedMigrations=543, elapsedMigrationTime=8983ms, totalElapsedMigrationTime=10667ms)
13:06:52,801  INFO || - [MasterJobContext] hz._hzInstance_26_jet.cached.thread-8 - [127.0.0.1]:5701 [jet] [3.1-SNAPSHOT] Start executing job '069c-611d-30b2-6555', execution 23e7-e3d4-1795-0863, execution graph in DOT format:
digraph DAG {
	"test";
}
```
Member 5703 attempted to migrate to 5704 before 5704 started. So the migration failed. Then 3 seconds later manual restart was requested and job was expected to start in 10 seconds, but started after 10.3 seconds.

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/ManualRestartTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/ManualRestartTest.java
@@ -90,7 +90,7 @@ public class ManualRestartTest extends JetTestSupport {
         JetInstance client = createJetClient();
         Job job = client.newJob(dag, new JobConfig().setAutoScaling(autoRestartOnMemberFailureEnabled));
 
-        assertTrueEventually(() -> assertEquals(NODE_COUNT, MockPS.initCount.get()), 10);
+        assertTrueEventually(() -> assertEquals(NODE_COUNT, MockPS.initCount.get()));
 
         // When the job is restarted after new members join to the cluster
         int newMemberCount = 2;
@@ -104,7 +104,7 @@ public class ManualRestartTest extends JetTestSupport {
 
         // Then, the job restarts
         int initCount = NODE_COUNT * 2 + newMemberCount;
-        assertTrueEventually(() -> assertEquals(initCount, MockPS.initCount.get()), 10);
+        assertTrueEventually(() -> assertEquals(initCount, MockPS.initCount.get()));
     }
 
     @Test
@@ -116,7 +116,7 @@ public class ManualRestartTest extends JetTestSupport {
         JetInstance client = createJetClient();
         Job job = client.newJob(dag);
 
-        assertJobStatusEventually(job, STARTING, 10);
+        assertJobStatusEventually(job, STARTING);
 
         // Then, the job cannot restart
         exception.expect(IllegalStateException.class);
@@ -175,7 +175,7 @@ public class ManualRestartTest extends JetTestSupport {
         sleepMillis(100);
         FailingMapStore.fail = true;
         job.restart();
-        assertTrueEventually(() -> assertTrue(FailingMapStore.failed), 5);
+        assertTrueEventually(() -> assertTrue(FailingMapStore.failed));
         FailingMapStore.fail = false;
 
         job.join();

